CONTRIBUTING.md

```diff
@@ -0,0 +1,62 @@
+# Contributing guide for Gregorio
+
+## Reporting an issue
+
+You can report a bug and request a new feature on the [bug tracker](https://github.com/gregorio-project/gregorio/issues).
+
+Please search for existing issues before reporting a new one. 
+
+Please do not use the issue tracker for personal support requests, instead use the mailing-list (https://gna.org/mail/?group=gregorio) or IRC: #gregorio on freenode.
+
+### Report a bug
+
+When you report a bug, please provide the following information:
+
+ * version of Gregorio
+ * a minimal example (gabc or gabc+tex if relevant) showing the bug
+ * a screenshot of the wrong result
+ * a precise description (if possible with a picture) of the expected output
+
+### Request feature
+
+When requesting a feature, please provide the following informations:
+
+ * a minimal example (gabc or gabc+tex) that currently doesn't work, but would
+   if the feature was done
+ * a precise description (if possible with a picture) of the expected output
+
+## Contributing code
+
+### Development environment
+
+It is not currently possible to compile Gregorio under Windows. The build requirements
+are the same under GNU/Linux and Mac OSX:
+
+ * [git](http://git-scm.com/)
+ * the [GNU Build System](http://airs.com/ian/configure/) (often referred to as "Autotools")
+ * [gcc](https://gcc.gnu.org/) (and associated tools)
+ * [GNU indent](https://www.gnu.org/software/indent/)
+
+### Coding style
+
+The code uses two spaces for indentation. The C code follows the [GNU coding standards](https://www.gnu.org/prep/standards/html_node/Writing-C.html).
+
+Use indent on your code before commiting it, with the `.indent.pro` file in repository's root folder: run `indent path/to/my/file.c` from the root directory.
+
```

Comment:
Okay, I have GNU indent installed and it worked fine.

I might point out, that it did make several changes to lib/characters.c.  Perhaps we need to run it on all files to update things?

Another thing, `indent` is only for the C files.  Do we want to follow the same indentation rules for the TeX files?  If so it might be helpful to have a link to [this page](http://stackoverflow.com/questions/2316677/can-git-automatically-switch-between-spaces-and-tabs) which shows how to implement an automatic change-over between tabs and spaces in git.